sagemaker-neo-notebooks/edge/cpp-integration/README.md

```diff
@@ -0,0 +1,216 @@
+
+#### Install nolhmann/json on Linux
+
+For linux based instruction, please check the following
+
+``` shell
+# checkout project and make install
+git clone https://github.com/nlohmann/json && \
+    cd json && \
+    cmake . && \
+    make && \
+    make install
+```
+
+#### Install nolhmann/json on Windows
+
+``` shell
+# checkout project
+git clone https://github.com/nlohmann/json
+cd json
+
+# build/install
+cmake .
+msbuild ALL_BUILD.vcxproj
+msbuild INSTALL.vcxproj /p:Configuration=Release
+```
+
+## Tutorial
+
+In this tutorial, we're going to compile a pre-train model through Neo compiler, then using DLR runtime for inference.
+
+### Linux Tutorial
+
+#### Install DLR on Linux
+
+Please follow this [link](https://neo-ai-dlr.readthedocs.io/en/latest/install.html#building-on-linux) to build DLR. Once done so, please locate libdlr.so in your build repository and copy to sagemaker-neo-notebooks/edge/windows/lib.
+
+
+Then you'll need to link libdlr.so to sagemaker-neo-notebooks/edge/windows/lib.
+
+``` shell
+cd sagemaker-neo-notebooks/edge/windows/lib
+ln -s neo-ai-dlr/build/lib/libdlr.so
+```
+
+#### Build tutorial program
+
+```shell
+# create a new build folder
+cd sagemaker-neo-notebooks/edge/windows
+mkdir build
+cd build
+
+# build through cmake
+cmake .. -DCMAKE_BUILD_TYPE=Release -DDLR_HOME="/home/ubuntu/neo-ai-dlr"
+make
+```
+
+#### Compile model on Linux
+
+1. To see how compilation work, just perform the following
+
+    ```shell
+    cd sagemaker-neo-notebooks/edge/windows/build/
+
+    #export your AWS access key
+    export AWS_ACCESS_KEY_ID="<your AWS_ACCESS_KEY_ID>"
+    export AWS_SECRET_ACCESS_KEY="<your AWS_SECRET_ACCESS_KEY>"
+    export AWS_SESSION_TOKEN="<your AWS_SESSION_TOKEN>"
+
+    # execute tutorial.exe to run Neo compiler
+    ./tutorial compile ml_c4
+    ```
+
+2. Once done so, unzip the compiled models for inference.
+
+   ``` shell
+   mkdir compiled_model
+   tar -zxvf ./compiled_model.tar.gz -C ./compled_model
+   ```
+
+3. To run the inference against the compiled model, just execute `./tutorial inference` to see inference against dog.npy
+
+### Windows Tutorial
+
+#### Install DLR on Windows
+
+Please follow instruction to install DLR.
+
+```
+cd neo-ai-dlr
+mkdir build
+cd build
+cmake .. -G"Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=Release -DCMAKE_CONFIGURATION_TYPES=Release
+```
+
+Then open neo-ai-dlr\dlr.sln in Visual Studio and click Build -> Build Solution. You will see dlr.dll and dlr.lib in the neo-ai-dlr\build\lib directory.
+
+#### Build tutorial to .exe
+
+```shell
+# create a new build folder
+cd sagemaker-neo-notebooks/edge/windows
+mkdir build
+cd build
+
+# build through cmake
+cmake ../ -DCMAKE_PREFIX_PATH="C:/Program Files (x86)/aws-cpp-sdk-all/lib/;C:/Program Files (x86)/aws-cpp-sdk-all/lib/cmake" -DBUILD_SHARED_LIBS=ON -DDLR_HOME="C:/neo-ai-dlr"
```

Comment:
Is it possible to use multi-line commands in windows?
```cmake .. \ 
-DCMAKE_PREFIX_PATH="C:/Program Files (x86)/aws-cpp-sdk-all/lib/;C:/Program Files (x86)/aws-cpp-sdk-all/lib/cmake" \
-DBUILD_SHARED_LIBS=ON \
-DDLR_HOME="C:/neo-ai-dlr"
```

Reply:
apparently, you can use back-tick in power-shell.